compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -500,7 +481,7 @@ fn link_staticlib<'a, B: ArchiveBuilder<'a>>(
 
         ab.add_archive(path, move |fname: &str| {
             // Ignore metadata files, no matter the name.
-            if fname == METADATA_FILENAME {
+            if fname == rustc_metadata::fs::METADATA_FILENAME {
```

Comment:
Could you import all necessary parts of `rustc_metadata::fs`?

Reply:
I fixed that: https://github.com/rust-lang/rust/pull/96544/commits/55ca8267a25ef3d5f7542594f4d3fc1da8a39a1c